syntax = "proto3";

package rag.v1;

option go_package = "Chimera-RAG/api/rag/v1;v1";

service LLMService {
  rpc AskStream (AskRequest) returns (stream AskResponse);
  rpc EmbedData (EmbedRequest) returns (EmbedResponse);

  // ğŸ”¥ æ–°å¢ï¼šè§£æå¹¶å‘é‡åŒ– PDF
  rpc ParseAndEmbed (ParseRequest) returns (ParseResponse);
}

message AskRequest {
  string query = 1;
  string session_id = 2;
  bool use_graph = 3;
}

message AskResponse {
  string answer_delta = 1;
  string thinking_log = 2;
}

message EmbedRequest {
  oneof data {
    string text = 1;
    string image_url = 2;
  }
}

message EmbedResponse {
  repeated float vector = 1;
}

// ğŸ”¥ æ–°å¢è¯·æ±‚ï¼šç›´æ¥ä¼ æ–‡ä»¶å†…å®¹çš„å­—èŠ‚æµ
message ParseRequest {
  bytes file_content = 1;
  string file_name = 2;
}

// ğŸ”¥ æ–°å¢å“åº”ï¼šè¿”å›å¤šä¸ªåˆ‡ç‰‡ï¼Œæ¯ä¸ªåˆ‡ç‰‡éƒ½æœ‰æ–‡æœ¬å’Œå‘é‡
message ParseResponse {
  repeated DocChunk chunks = 1;
}

message DocChunk {
  string content = 1;       // åˆ‡åˆ†åçš„æ–‡æœ¬ç‰‡æ®µ
  repeated float vector = 2; // è¯¥ç‰‡æ®µå¯¹åº”çš„ 384ç»´ å‘é‡
  int32 page_number = 3;    // é¡µç  (ç”¨äºå‰ç«¯è·³è½¬)
}